core/libp2p/network/network.hpp

```diff
@@ -8,6 +8,7 @@
 
 #include "libp2p/network/dialer.hpp"
 #include "libp2p/network/listener.hpp"
+#include "libp2p/event/bus.hpp"
 
 namespace libp2p::network {
 
```

Comment:
As this is a PR, introducing the Network finally, please brush this class - add normal docs, etc.

Also, can't you already inject a bus and emit some of the events through `Listener` and `Dialer`? This is a pretty small PR, so I think it's a good place to do it